src/Select.js

```diff
@@ -1,238 +1,116 @@
-        }
+    handleClear = e => {
+        e.stopPropagation()
+        this.props.onChange({})
     }
 
     render() {
         const {
-            dense,
-            disabled,
-            onChange,
-            value,
             children,
-            name,
             tabIndex,
-            valid,
-            warning,
-            error,
-            loading,
-            className,
+            selected,
+            placeholder,
+            clearable,
         } = this.props
-
-        const { focus } = this.state
+        const { open } = this.state
 
         return (
             <div
-                className={cx('select', className, {
-                    dense,
-                    disabled,
-                    focus,
-                    valid,
-                    warning,
-                    error,
-                })}
+                className="container"
+                ref={this.containerRef}
+                tabIndex={tabIndex}
+                onFocus={this.handleFocus}
+                onClick={this.handleOpen}
             >
-                <select
-                    onChange={onChange}
-                    value={value || -1}
-                    disabled={disabled}
-                    onFocus={this.onFocus}
-                    onBlur={this.onBlur}
-                    ref={this.selectRef}
-                    name={name}
-                    tabIndex={tabIndex}
-                >
-                    <option
-                        key="hidden-default-value"
-                        hidden
-                        disabled
-                        value="-1"
-                    />
-                    {children}
-                </select>
-
-                <div className="status-icon">
-                    <TailIcon />
-
-                    <StatusIconNoDefault
-                        error={error}
-                        valid={valid}
-                        loading={loading}
-                        warning={warning}
-                    />
-                </div>
-                <style jsx>{styles}</style>
+                <SelectInput
+                    placeholder={placeholder}
+                    selected={selected}
+                    clearable={clearable}
+                    onClear={this.handleClear}
+                    open={open}
+                />
+                {open && (
+                    <SelectDropdown
+                        onClick={this.handleOptionClick}
+                        selected={selected}
+                    >
+                        {children}
+                    </SelectDropdown>
+                )}
+
+                <style jsx>{`
+                    .container {
+                        position: relative;
+                        outline: 0;
+                    }
+                `}</style>
             </div>
         )
     }
 }
 
-Select.propTypes = {
-    className: propTypes.string,
+Select.defaultProps = {
+    clearable: false,
```

Comment:
Not needed, not providing boolean will result in a falsy value anyways (we don't do that in other components either unless the default value is true)

Reply:
Yeah that's something I'm used to doing from having this rule enabled in a lot of codebases I've worked on: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md. I think the reasoning behind it makes sense. If we don't do it elsewhere I'll remove it, though I wouldn't mind enabling this rule (only with function props I find it a bit superfluous).

Reply:
Yes, I'd prefer that as well. Not using default props was a design decision by @varl .
For now, I think you can remove it, if we want to introduce that rule, it should be a separate PR

Reply:
Sounds good, I'll do that 👍

Reply:
Ok fixed.